tests/test_profiles.py

```diff
@@ -240,3 +297,127 @@ def test_profile_conversion(self):
+            sym=True,
+        )
+        eq1.solve(
+            constraints=get_fixed_boundary_constraints(kinetic=False),
+            objective=ObjectiveFunction(objectives=ForceBalance()),
+            maxiter=5,
+        )
+        eq2 = Equilibrium(
+            electron_temperature=Te,
+            electron_density=ne,
+            atomic_number=1,
+            ion_temperature=Ti,
+            iota=PowerSeriesProfile([1.61]),
+            Psi=np.pi,  # so B ~ 1 T
+            NFP=1,
+            L=LM_resolution,
+            M=LM_resolution,
+            N=0,
+            L_grid=2 * LM_resolution,
+            M_grid=2 * LM_resolution,
+            N_grid=0,
+            sym=True,
+        )
+        eq2.solve(
+            constraints=get_fixed_boundary_constraints(kinetic=True),
+            objective=ObjectiveFunction(objectives=ForceBalance()),
+            maxiter=5,
+        )
+        np.testing.assert_allclose(eq1.R_lmn, eq2.R_lmn, atol=1e-14)
+        np.testing.assert_allclose(eq1.Z_lmn, eq2.Z_lmn, atol=1e-14)
+        np.testing.assert_allclose(eq1.L_lmn, eq2.L_lmn, atol=1e-14)
+
+    @pytest.mark.unit
+    def test_kinetic_pressure(self):
+        """Test that both ways of computing pressure are equivalent."""
+        ne = PowerSeriesProfile(3.0e19 * np.array([1, -1]), modes=[0, 10])
+        Te = PowerSeriesProfile(2.0e3 * np.array([1, -1]), modes=[0, 2])
+        Ti = Te
+        pressure = elementary_charge * (ne * Te + ne * Ti)
+        print("pressure params:", pressure.params)
+
+        LM_resolution = 6
+        eq1 = Equilibrium(
+            pressure=pressure,
+            iota=PowerSeriesProfile([1.61]),
+            Psi=np.pi,  # so B ~ 1 T
+            NFP=1,
+            L=LM_resolution,
+            M=LM_resolution,
+            N=0,
+            L_grid=2 * LM_resolution,
+            M_grid=2 * LM_resolution,
+            N_grid=0,
+            sym=True,
+        )
+        eq2 = Equilibrium(
+            electron_temperature=Te,
+            electron_density=ne,
+            iota=PowerSeriesProfile([1.61]),
+            Psi=np.pi,  # so B ~ 1 T
+            NFP=1,
+            L=LM_resolution,
+            M=LM_resolution,
+            N=0,
+            L_grid=2 * LM_resolution,
+            M_grid=2 * LM_resolution,
+            N_grid=0,
+            sym=True,
+        )
+        grid = LinearGrid(L=20)
+        data1 = eq1.compute(["p", "p_r"], grid=grid)
+        data2 = eq2.compute(["p", "p_r"], grid=grid)
+
+        assert np.all(np.isnan(data1["ne"]))
+        assert np.all(np.isnan(data1["Te"]))
+        assert np.all(np.isnan(data1["Ti"]))
+        assert np.all(np.isnan(data1["Zeff"]))
+        assert np.all(np.isnan(data1["ne_r"]))
+        assert np.all(np.isnan(data1["Te_r"]))
+        assert np.all(np.isnan(data1["Ti_r"]))
+        assert np.all(np.isnan(data1["Zeff_r"]))
+        assert np.all(data2["Te"] == data2["Ti"])
+        assert np.all(data2["Te_r"] == data2["Ti_r"])
```

Comment:
No similar tests for density?

Reply:
If the temperatures and pressures are the same I don't think we need an explicit test for density. And it would be a pretty trivial test, just `ne==ne` since there's no extra math involved, its just a pass-through.